File: easy/maxProduct.py
class Solution:
    def maximumProduct(self, nums: List[int]) -> int:
        nums.sort()
        guan = (nums[0]*nums[1],nums[-2]*nums[-3])
        return max(guan) * nums[-1] if nums[-1]>0 else min(guan)*nums[-1]
"""Given an integer array nums, find three numbers whose product is maximum and return the maximum product."""

File: easy/max_product.rs
impl Solution {
    pub fn max_product(mut n: i32) -> i32 {
        let mut max = 0;
        let mut best = 0;
        while n != 0{
            best = best.max((n%10) * max);
            max = max.max(n%10);
            n /= 10;
        }
        best


    }
}
// You are given a positive integer n.

// Return the maximum product of any two digits in n.

// Note: You may use the same digit twice if it appears more than once in n.

 
